crates/interpreter/src/instructions/system.rs

```diff
@@ -179,6 +157,29 @@ pub fn returndataload<H: Host + ?Sized>(interpreter: &mut Interpreter, _host: &m
     *offset = B256::from(output).into();
 }
 
+// common logic for copying data from a source buffer to the EVM's memory
+fn copy_to_memory(
+    interpreter: &mut Interpreter,
+    memory_offset: U256,
+    data_offset: usize,
+    len: usize,
+    source: *const [u8],
+) {
```

Comment:
I would return a true/false here. Otherwise function silently fails, and can work only if it is in last line of instruction.